src/OSPSuite.Core/Domain/Services/SpatialStructureMerger.cs

```diff
@@ -90,32 +91,32 @@ private void createMergedContainerStructureInRoot(IContainer root, SimulationBui
 
       private Func<IContainer, IContainer> mapContainerDef(SimulationBuilder simulationBuilder) => container => _containerMapper.MapFrom(container, simulationBuilder);
 
-      private Action<IContainer> mergeTopContainerInStructure(IContainer root) => topContainer =>
+      private Action<IContainer, MergeBehavior> mergeTopContainerInStructure(IContainer root) => (topContainer, mergeBehavior) =>
       {
          //probably should never happen
          if (topContainer == null)
             return;
 
          //In this case, we add or replace the top container
          if (topContainer.ParentPath == null || string.IsNullOrEmpty(topContainer.ParentPath.PathAsString))
-            addOrReplaceContainer(topContainer, root);
+            addOrReplaceContainer(topContainer, root, mergeBehavior);
          else
-            insertTopContainerIntoStructure(topContainer, root);
+            insertTopContainerIntoStructure(topContainer, root, mergeBehavior);
       };
 
-      private void insertTopContainerIntoStructure(IContainer topContainer, IContainer root)
+      private void insertTopContainerIntoStructure(IContainer topContainer, IContainer root, MergeBehavior mergeBehavior)
       {
          var parentContainer = topContainer.ParentPath.Resolve<IContainer>(root);
          if (parentContainer == null)
             throw new OSPSuiteException(Error.CannotFindParentContainerWithPath(topContainer.ParentPath.PathAsString, topContainer.Name));
 
-         addOrReplaceContainer(topContainer, parentContainer);
+         addOrReplaceContainer(topContainer, parentContainer, mergeBehavior);
       }
 
-      private void addOrReplaceContainer(IContainer containerToAdd, IContainer parentContainer)
+      private void addOrReplaceContainer(IContainer containerToAdd, IContainer parentContainer, MergeBehavior mergeBehavior)
       {
-         //we do have a special case to deal with when dealing with MoleculeProperties container that we merge instead of replacing
-         if (containerToAdd.IsNamed(Constants.MOLECULE_PROPERTIES))
+         //Merge behavior is merged or we have a special case to deal with when dealing with MoleculeProperties container that we merge instead of replacing
```

Comment:
Did you mean 'Merge behavior is extend' instead of 'merged'

Reply:
yep